clippy_lints/src/redundant_async_block.rs

```diff
@@ -82,3 +87,33 @@ impl<'ast> AstVisitor<'ast> for AwaitDetector {
         }
     }
 }
+
+/// Check whether an expression may have captured a local variable
+/// by reference. This is done by looking for paths with only one
+/// segment, except as a prefix of `.await` since this would
+/// be captured by value.
+///
+/// There will be false positives, for example when calling a local
+/// function, but at the AST level this cannot be distinguished from
+/// calling a captured closure from the local environment.
```

Comment:
Could you give an example when this is a false positive? From the explanation, it sounds like detecting `function` as a ref path, would cause false negatives and not the other way around :)

Reply:
The false positives comment concerns the function itself, I'll be clearer in the comment. Since this function is used to prevent the lint from triggering, this causes false negatives at the lint level.

I will avoid using "false …" at the function level.